buffer/src/main/java/io/netty/buffer/DuplicatedByteBuf.java

```diff
@@ -31,18 +31,31 @@
  */
 public class DuplicatedByteBuf extends AbstractDerivedByteBuf {
 
-    private final ByteBuf buffer;
+    private ByteBuf buffer;
 
     public DuplicatedByteBuf(ByteBuf buffer) {
         super(buffer.maxCapacity());
+        init(buffer);
+    }
+
+    /**
+     * Special constructor for sub-classes. Be aware that {@link #init(ByteBuf)} needs to be called after
+     * construction.
+     */
+    DuplicatedByteBuf() {
+        super(Integer.MAX_VALUE);
+    }
+
+    final void init(ByteBuf buffer) {
+        maxCapacity(buffer.maxCapacity());
 
-        if (buffer instanceof DuplicatedByteBuf) {
+        if (buffer instanceof DuplicatedByteBuf && !(buffer instanceof PooledDuplicatedByteBuf)) {
```

Comment:
The class usage of class hierarchy is a bit inverted here. The base class knows about an inheriting class. Can we break this relationship?  Maybe change the signature of init to `final void init(ByteBuf buffer, boolean unwrapDuplicateBuffer)`?